tests/v2.test.js

```diff
@@ -0,0 +1,443 @@
+});
+
+const root = pathTool.resolve(__dirname, '../');
+
+beforeAll(async () => {
+  const version = await api.post('version')
+    .then(response => JSON.parse(response).number);
+
+  await Promise.all([
+    api.post(`version/${version}/backend`).form({
+      hostname: `${process.env.S3_BUCKET_NAME}.s3.amazonaws.com`,
+      name: 'bucket',
+      shield: 'dca-dc-us',
+    }),
+    api.post(`version/${version}/condition`).form({
+      name: 'cats-prefix',
+      statement: 'req.http.X-IIIF-Prefix == "kittehs"',
+      type: 'REQUEST',
+    }).then(() => api.post(`version/${version}/header`).form({
+      name: 'cats-prefix',
+      type: 'request',
+      request_condition: 'cats-prefix',
+      action: 'set',
+      dst: 'url',
+      src: '"/cats/" req.http.X-IIIF-Identifier "?" req.url.qs',
+    })),
+    api.post(`version/${version}/domain`).form({
+      name: domain,
+    }),
+    api.post(`version/${version}/header`).form({
+      name: 'vary-test-run',
+      type: 'cache',
+      action: 'append',
+      dst: 'http.Vary',
+      src: '"X-Test-Run"',
+    }),
+    api.post(`version/${version}/snippet`).form({
+      name: 'IIIF config',
+      dynamic: 0,
+      type: 'init',
+      content: `
+sub iiif_config {
+
+  set req.http.X-IIIF-Version = if(req.http.X-Test-IIIF-Version, req.http.X-Test-IIIF-Version, "2");
+
+}
+`,
+    }),
+    fs.readFile(`${root}/iiif.vcl`)
+      .then((contents) => {
+        api.post(`version/${version}/vcl`).form({
+          name: 'IIIF',
+          content: contents,
+          main: true,
+        });
+      }),
+  ]);
+
+  return api.put(`version/${version}/activate`);
+});
+
+const imageUri = (originalParts) => {
+  const parts = Object.assign({
+    prefix: '',
+    identifier: 'pug-life.jpg',
+    region: 'full',
+    size: 'full',
+    rotation: 0,
+    quality: 'default',
+    format: 'jpg',
+  }, originalParts);
+
+  return `${parts.prefix}/${parts.identifier}/${parts.region}/${parts.size}/${parts.rotation}/${parts.quality}.${parts.format}`;
+};
+
+describe('Image request', () => {
+  const ok = (iiifParameters, requestedIoParameters = {}, method = 'GET') => {
+    const ioParameters = Object.assign({
+      format: 'pjpg',
+    }, requestedIoParameters);
+
+    const ioQueryParameters = new URLSearchParams(ioParameters);
+
+    expect.assertions(2);
+
+    return http({
+      method,
+      uri: imageUri(iiifParameters),
+    })
+      .then((response) => {
+        expect(response.statusCode).toBe(200);
+        expect(response.headers['x-fastly-io-url']).toBe(`/pug-life.jpg?${ioQueryParameters.toString()}`);
+      });
+  };
+
+  const error = (iiifParameters, statusCode, method = 'GET') => {
+    expect.assertions(1);
+
+    return http({
+      method,
+      uri: imageUri(iiifParameters),
+    })
+      .catch(exception => expect(exception).toHaveProperty('statusCode', statusCode));
+  };
+
+  const badRequest = iiifParameters => error(iiifParameters, 400);
```

Comment:
You probably already know this, and it's not a problem here, but pointing it out as it's a common error: code in `describe` is executed immediately when the file is loaded. Illustration https://gist.github.com/tamlyn/dd6d54cdbe6eccd51a4bc61f5844bec4

Reply:
Wasn't aware of that! Doesn't look a problem here, but any alternative pattern you'd recommend?

Reply:
My rule of thumb is that initialising a variable or declaring a function is fine in a describe but actually calling a function should go in a `before`.